cogboard-webapp/src/components/widgets/types/LogViewerWidget/Toolbar/QuarantineModal/index.js

```diff
@@ -33,6 +34,25 @@ const QuarantineModal = ({ wid, quarantine }) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [similarLogs.quarantine]);
 
+  const isChecked = (checked, endTimestamp) => {
+    if (endTimestamp) {
+      const inFuture = moment.utc(endTimestamp * 1000).local() > moment();
+      return checked && inFuture;
+    }
+    return checked;
+  };
+
+  const toggleChecked = rule => {
+    const endTimestamp = rule.endTimestamp;
+    if (endTimestamp) {
+      const inFuture = moment.utc(endTimestamp * 1000).local() > moment();
```

Comment:
co my tutaj sprawdzamy? trochę nie rozumiem tej logiki, ale z kontekstu domyślam się, że kiedyś podaliśmy endTimestamp tworząc/modyfikując jakąś regułę X i teraz ta reguła zdążyła się już wyłączyć.  I teraz jeśli ponownie ją włączymy, to ta logika odpowiada za wywalenie endTimestamp, bo jest niepotrzebny, dopóki go nie ustawimy znowu na jakiś nowy timestamp. Jeśli dobrze myślę, to proszę o bardziej sensowne zorganizowanie tego kodu, bo sporo czasu spędziłem na ogarnięciu tego